pages/api/student_email_join.js

```diff
@@ -1,37 +1,63 @@
 import prisma from '../../prisma/prisma';
+import { unstable_getServerSession } from 'next-auth';
+import { authOptions } from './auth/[...nextauth]';
 
 export default async function handle(req, res) {
-  const data = JSON.parse(req.body);
-  const userInfoReq = await fetch(
-    `http://localhost:3002/getstudentprofile?email=${data['email']}`
-  );
-  const userInfo = await userInfoReq.json();
-  if (userInfo.length !== 0) {
-    const existingStudent = await prisma.classroom.findMany({
-      where: {
-        classroomId: data['classId'][0]
-      }
-    });
-    const ids = existingStudent[0]['fccUserIds'];
-    if (ids.includes(userInfo[0]['uuid'])) {
-      return res.status(409).json({
-        error: 1,
-        msg: 'Account already exists'
-      });
+  // unstable_getServerSession is recommended here: https://next-auth.js.org/configuration/nextjs
+  const session = await unstable_getServerSession(req, res, authOptions);
+
+  if (!req.method == 'PUT') {
+    res.status(405).end();
+  }
+
+  if (!session) {
+    res.status(403).end();
+  }
+
+  const body = req.body;
+  // Grab user info here
+  const userInfo = await prisma.user.findUnique({
+    where: {
+      email: session.user.email
+    },
+    select: {
+      id: true
+    }
+  });
+  // Grab class info here
+  const checkClass = await prisma.classroom.findUniqueOrThrow({
+    where: {
+      classroomId: body.join[0]
+    },
+    select: {
+      fccUserIds: true
     }
-    const createdStudentEmail = await prisma.classroom.update({
+  });
+  const existsInClassroom = checkClass.fccUserIds.includes(userInfo.id);
+  if (existsInClassroom) {
+    res.status(409).end();
+  } else if (userInfo.role === 'TEACHER' || userInfo.role === 'ADMIN') {
+    // update the teachers to also include (not overwrite) this new teacher.
+    // some prisma query to add a new teacher to the list of teachers in that classroom.
```

Comment:
Having an empty if statement body here seems odd.  It looks like there is some additional logic you intend to add here in the future.  Can you file an issue for this additional logic and somehow remove the empty if statement for the time being?

Reply:
Sure thing. sorry this took a while for me to get back to, just pushed the change and will file an issue right now.